mat64/symmetric.go

```diff
@@ -112,8 +119,12 @@ func (s *SymDense) reuseAs(n int) {
 			Data:   use(s.mat.Data, n*n),
 			Uplo:   blas.Upper,
 		}
+		s.cap = n
 		return
 	}
+	if s.mat.N > s.cap {
```

Comment:
Move this above the assignment to *s, as written this will never fail. The intention is to catch a case where something previously incorrectly set cap (or failed to set it at all more likely), rather than that the current reuseAs called failed.